pkgs/unified_analytics/lib/src/event.dart

```diff
@@ -887,3 +848,15 @@ final class Event {
     }
   }
 }
+
+/// A base class for custom metrics that will be defined by a unified_analytics
+/// client.
+///
+/// This base type can be used as a parameter in any event constructor that
+/// allows custom metrics to be added by a unified_analytics client.
+abstract base class CustomMetrics {
+  /// Converts the custom metrics data to a [Map] object.
+  /// 
+  /// This must be a JSON encodable [Map].
+  Map<String, Object?> toMap();
```

Comment:
Can we make this `Map<String, Object>` instead if we're already disallowing `null`?